src/plays/typing-speed-test/components/Main.js

```diff
@@ -0,0 +1,181 @@
+    setActiveWordIndex((index) => index + 1);
+    setUserInput("");
+
+    // To check each character and word
+    let wordToCompare = words[activeWordIndex];
+    let isWordMatch = wordToCompare === value.trim();
+
+    setTypedWordsArray([...typedWordsArray, isWordMatch]);
+    if (isWordMatch) {
+      setCorrectWords(correctWords + 1);
+      setCorrectCharLength((charLen) => charLen + value.trim().length);
+    } else {
+      setIncorrectWords(incorrectWords + 1);
+    }
+  };
+
+  // Hanlde user input
+  const handleUserInput = (value) => {
+
+    if (!isTimerStart) {
+      setIsTimerStart(true);
+      setStatus("started");
+    }
+
+    if (activeWordIndex === words.length) return;
+
+    if (value.endsWith(" ")) {
+      checkIsWordMatch(value);
+
+      // Check if length of activeWordIndex  === words then end the test
+      if (activeWordIndex === words.length - 1) {
+        setStatus("finished");
+        setTimer(60);
+        setIsResultModalOpen(true);
+      }
+    } else {
+      setUserInput(value);
+    }
+  };
+
+  useEffect(() => {
+    if (timer === 0) {
+      setStatus("finished");
+      setIsResultModalOpen(true);
+      return;
+    }
+
+    // To start countdown
+    let timerTimout = setTimeout(() => {
+      if (status === "started") {
+        setTimer(timer - 1);
+      }
+    }, 1000);
+
+    return () => clearTimeout(timerTimout);
+  }, [timer, status]);
+
+  useEffect(() => {
+    // Setting typing text
+    setWords(generateWords());
+
+    // Setting focus on input
+    userInputRef.current.focus();
+  }, []);
+
+  return (
+    <>
+      <div className=" max-w-3xl text-center my-8 mx-auto flex flex-col justify-center">
+        <h2 className="text-[1.6rem] md:text-4xl  text-violet-600 font-bold">
+          Typing Speed Test ⌨️🚀
+        </h2>
+        <p className="text-base my-2 md:text-lg md:my-3 text-gray-600  ">
+          Test your typing skills
+        </p>
+
+        {/* Statistics & Timer */}
+        <div className="flex flex-col justify-around items-center my-5 sm:flex-row md:my-6 md:mt-8 ">
+          <Timer timer={timer} />
+          <Stats
+            wpm={correctWords}
+            cpm={correctCharLength}
+            accuracy={Math.round(
+              (correctWords / (correctWords + incorrectWords)) * 100
+            )}
+          />
+        </div>
+
+        <div className="max-w-3xl my-6 mx-auto leading-9 text-justify md:leading-10 tracking-wide">
+          {words.length ? (
+            words.map((word, index) => (
+              <Word
+                key={index}
+                text={word}
+                isActive={index === activeWordIndex}
+                correct={typedWordsArray[index]}
+              />
+            ))
+          ) : (
+            <p>Text loading please wait ...</p>
+          )}
+        </div>
+
+        <div className="flex items-center justify-center mt-8 space-x-7">
+          <input

```

Comment:
Can we use textarea here for better UX? as the input box will be too small?

Reply:
@makdoom , Here we are asking the user to type the given para right and that small input box will not be good, so we should change that input to textarea.

Reply:
Done